tools/shared.py

```diff
@@ -1404,6 +1408,21 @@ def map(self, func, tasks):
             for t in tasks:
               results += [func(t)]
             return results
+
+          def map_async(self, func, tasks, **kwargs):
+            class Result:
+              def __init__(self, func, tasks):
+                self.func = func
+                self.tasks = tasks
+
+              def get(self, timeout):
+                results = []
+                for t in tasks:
+                  results += [func(t)]
+                return results
+
+            return Result(func, tasks)
```

Comment:
What is this new function?  Looks like it might not be used?

Reply:
It's used when we use the multiprocessing pool, we call `map_async` there. This code path is used when the number of cores is 1, instead of the native python actual parallel multiprocessing code.

(This was crashing before this PR - we just didn't handle the single-core case for a long time, which I guess shows that none of our developers are on single-core machines...)